### PR TITLE
Nits

### PR DESCRIPTION
**Pull request by connor4312**
Some minor tweaks that I thought of overnight. One per commit.

 - Added a simple mocking implementation that can make testing in certain scenarios easier.
 - Fixed a case where we would likely end up backing off more than we should.
 - Make `key()` chainable, whereas in the past someone could _only_ request the keys as strings.
 - Allowed clients to set the encoding in  the `string()` and `strings()` methods

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,12 +2,14 @@
   "name": "etcd3",
   "version": "0.1.0",
   "description": "Node client for etcd3",
-  "main": "dist/index.js",
+  "main": "lib/src/index.js",
```
Comment (SimonSchick) [inline package.json]: typings?

File: src/index.ts [hunk 2]
```diff
@@ -62,6 +62,23 @@ export class Etcd3 {
...
+   */
+  public mock<T extends RPC.ICallable>(callable: T): T {
+    this.pool.mock(callable);
+    return callable;
+  }
+
+  /**
+   * Removes any previously-inserted mock.
+   */
+  public unmock(): void {
```
Comment (SimonSchick) [inline src/index.ts]: If it's void why return?

File: src/kv-builder.ts [hunk 3]
```diff
@@ -213,34 +213,36 @@ export class MultiRangeBuilder extends RangeBuilder {
...
+  public keys(): this {
     this.request.keys_only = true;
-    return this.exec().then(res => res.kvs.map(kv => kv.key.toString()));
+    return this;
   }
 
   /**
    * Runs the built request and parses the returned keys as JSON.
    */
   public json(): Promise<any> {
```
Comment (SimonSchick) [inline src/kv-builder.ts]: Probably `object`?

File: src/kv-builder.ts [hunk 4]
```diff
@@ -213,34 +213,36 @@ export class MultiRangeBuilder extends RangeBuilder {
...
-    return this.exec().then(res => res.kvs.map(kv => JSON.stringify(kv.value)));
+    return this.buffers().then(res => res.map(value => JSON.stringify(value)));
```
Comment (SimonSchick) [inline src/kv-builder.ts]: Can't you just pass `JSON.stringify` directly?

File: test/shared-pool.test.ts [hunk 5]
```diff
@@ -65,4 +65,12 @@ describe('shared pool', () => {
     clock.tick(500);
     expect(await getAll()).to.deep.equal([0, 2, 1]);
   });
+
+  it('should not back off multiple times if multiple callers fail', async () => {
+    const cnx = await pool.pull();
+    pool.fail(cnx);
+    expect((<any> pool).resources[0].availableAfter).to.equal(Date.now() + 500);
```
Comment (SimonSchick) [inline test/shared-pool.test.ts]: Could dedupe a little.

File: src/kv-builder.ts (hunk 4 above)
Comment (connor4312) [inline src/kv-builder.ts]: stringify takes multiple arguments, I don't really feel comfortable with that

File: src/index.ts (hunk 2 above)
Comment (connor4312) [inline src/index.ts]: Fixed!

File: package.json (hunk 1 above)
Comment (connor4312) [inline package.json]: Fixed!

File: src/index.ts (hunk 2 above)
Comment (SimonSchick) [inline src/index.ts]: But, it's not?

Comment (connor4312) [inline src/index.ts]: forgot to save, there we go

**Pull request merged**